core/src/main/java/com/sequenceiq/cloudbreak/service/rdsconfig/DbUsernameConverterService.java

```diff
@@ -0,0 +1,47 @@
+package com.sequenceiq.cloudbreak.service.rdsconfig;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DbUsernameConverterService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DbUsernameConverterService.class);
+
+    public String toConnectionUsername(String host, String dbUser) {
+        if (DbHostnameMatcher.isAzure(host)) {
+            LOGGER.debug("Detected Azure database server {}, appending short hostname to connection username", host);
+            return dbUser + "@" + host.substring(0, host.indexOf('.'));
```

Comment:
this could also be simplified with `StringUtils.substringBefore`

Reply:
Thanks, I added it.